pkg/bitcoin/electrum/electrum_integration_test.go

```diff
@@ -0,0 +1,343 @@
+//go:build integration
+// +build integration
+
+package electrum
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/go-test/deep"
+	"golang.org/x/exp/slices"
+
+	"github.com/keep-network/keep-core/pkg/bitcoin"
+)
+
+// TODO: Include integration test in the CI.
+// To run the tests execute `go test -v -tags=integration ./...`
```

Comment:
Also, I'm getting those kind of errors:
```
--- FAIL: TestGetLatestBlockHeight_Integration (2.34s)
    --- FAIL: TestGetLatestBlockHeight_Integration/electrs-esplora_ssl (0.31s)
        electrum_integration_test.go:290: failed to initialize electrum client: [EOF]
--- FAIL: TestGetBlockHeader_Integration (3.09s)
    --- FAIL: TestGetBlockHeader_Integration/electrs-esplora_tcp (1.15s)
        electrum_integration_test.go:290: failed to get server version: [retry timeout [1s] exceeded; most recent error: [ServerVersion failed: [request timeout]]]
    --- FAIL: TestGetBlockHeader_Integration/electrs-esplora_ssl (0.31s)
        electrum_integration_test.go:290: failed to initialize electrum client: [EOF]
--- FAIL: TestGetBlockHeader_Negative_Integration (4.65s)
    --- FAIL: TestGetBlockHeader_Negative_Integration/electrs-esplora_tcp (1.16s)
        electrum_integration_test.go:290: failed to get server version: [retry timeout [1s] exceeded; most recent error: [ServerVersion failed: [request timeout]]]
    --- FAIL: TestGetBlockHeader_Negative_Integration/electrs-esplora_ssl (0.31s)
        electrum_integration_test.go:290: failed to initialize electrum client: [read tcp 192.168.2.102:60812->35.225.54.191:60002: read: connection reset by peer]
```

Reply:
Unfortunately, the connection to servers is not very stable. 
For the requests, we've got retries, which work pretty well. But we still need to add retries for connection establishment and reconnects on connection drops. I left TODOs in the code:
https://github.com/keep-network/keep-core/blob/79a8cdb39aa96a7911ea05c0f9380a1f98243df0/pkg/bitcoin/electrum/electrum.go#L48
https://github.com/keep-network/keep-core/blob/79a8cdb39aa96a7911ea05c0f9380a1f98243df0/pkg/bitcoin/electrum/electrum.go#L116

We will handle those in a follow-up PR.

Please try re-running the electrum integration tests and see if you will be able to get the failing tests passed at some point.
```
go test -v -tags=integration ./pkg/bitcoin/electrum
```

Reply:
I ran the tests, but some of them still fail.
Hope the connection retires will solve the problem.